guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,927 @@
+      .load::<PostsWithUserName>(&connection);
+    println!("{:?}", joined); 
+    // > [PostsWithUserName { user_name: UserName { first_name: "Gordon", last_name: "Freeman", full_name: "GordonFreeman" }, title: "Thoughts on Tomorrow's Experiment", content: "What could possibly go wrong? }]
+}
+```
+
+If we were to forget one of `QueryableByName`'s required struct annotations,
+we would see the following error.
+
+```rust
+error: Your struct must either be annotated with `#[table_name = "foo"]` or have all of its fields annotated with `#[sql_type = "Integer"]`
+ --> src/models.rs:4:17
+  |
+4 | #[derive(Debug, QueryableByName)]
+  |                 ^^^^^^^^^^^^^^^
+```
+
+If we were to comment out the `sql_type` annotation for `UserName`'s `full_name` field,
+we would see the folliwing error.
+
+```rust
+error[E0412]: cannot find type `full_name` in module `users`
+  --> src/models.rs:12:17
+   |
+12 | #[derive(Debug, QueryableByName)]
+   |                 ^^^^^^^^^^^^^^^ not found in `users`
+```
+
+## Insertable
+
+The typical way to insert data in Diesel is by working with tuples.
+However, this can become tedious when inserting a lot of data,
+especially if it were a web form deserialized by a library like Serde.
+The `Insertable` trait is meant to be implemented on structs for 
+inserting this deserialized web form data.
+To implement `Insertable` on your struct,
+add the [`#[derive(Insertable)]`][insertable_doc] trait annotation.
+
+As with typical Diesel inserts, 
+you will still be using the [`.insert_into()`]
+method to generate a SQL `INSERT` statement for that table.
+You may chain [`.values()`] or [`default_values()`]
+to add the values for that `INSERT` statement.
+[`.values()`] accepts a reference to an `Insertable` struct for inserting a single record.
+
+You may also pass [`.values()`] a `&Vec<T>` or a `&[T]` to insert multiple records at once.
+On backends that support the `DEFAULT` keyword,
+the data will be inserted in a single query.
+On SQLite, one query will be performed per row.
+
+[`.insert_into()`]: https://docs.diesel.rs/diesel/fn.insert_into.html
+[`.values()`]: https://docs.diesel.rs/diesel/query_builder/struct.IncompleteInsertStatement.html#method.values
+[`.default_values()`]: https://docs.diesel.rs/diesel/query_builder/struct.IncompleteInsertStatement.html#method.default_values
+[insertable_doc]: https://docs.diesel.rs/diesel/prelude/trait.Insertable.html
+
+For `Insertable` structs, Diesel needs to know the corresponding table name.
+You must add the `#[table_name="some_table_name"]` annotation to your `Insertable` struct.
+If your struct has different field names than the columns they reference,
+they may be annotated with `#[column_name(some_column_name)]`.
+
+Typically, you will not use `Queryable` and `Insertable` together,
+as a web form for a new record wouldn't have such fields as
```

Comment:
I'd split this into two sentences since web form are just _one_ use case (and no guide currently does anything with web form IIRC).

> Typically, you will not use `Queryable` and `Insertable` together;
> Thinking of web forms again, for a new record the form wouldn't have such fields as `id`, `created_at`, or `updated_at`.